docs/resources/server/enchantments/index.md

```diff
@@ -0,0 +1,307 @@
+    ENCHANTMENT_COMPONENT_TYPES.register("increment",
+        () -> DataComponentType.<ConditionalEffect<Increment>>builder()
+            .persistent(ConditionalEffect.codec(Increment.CODEC, LootContextParamSets.EMPTY))
+            .build());
+```
+
+```java
+// Somewhere in game logic where an `itemStack` is available.
+// `unmodifiedValue` is an integer.
+MutableInt mutableValue = new MutableInt(unmodifiedValue);
+EnchantmentHelper.runIterationOnItem(itemStack, (enchantmentHolder, enchantLevel) -> Enchantment.applyEffects(
+    // Isolates the ConditionalEffect<Increment> from the provided holder and wraps it in a list for applyEffects.
+    enchantmentHolder.value().getEffects(INCREMENT.value()),
+
+    // Produces a LootContext. Other context helpers from the Enchantment class
+    // include itemContext, locationContext, entityContext, and blockHitContext.
+    Enchantment.damageContext(server, enchantLevel, target, damageSource), 
+    
+    // Runs for each successful <ConditionalEffect<T>>.
+    // `exampleData` is an Increment instance.
+    // This line actually performs the value adjustment.
+    // Each time it runs, mutableValue is set to (exampleData's value) + mutableValue + enchantLevel.
+    exampleData -> mutableValue.setValue(exampleData.add(mutableValue.getValue()))
+));
+
+// Use mutableValue elsewhere in your game logic.
+```
+
+We define an enchantment effect component called `INCREMENT` as `List<ConditionalEffect<Increment>>`. The `Increment` object it contains is a wrapper around an integer that defines a method `add(int x)`, which adds its internal value to the provided integer and returns the result. Imagine that you want to use this object to increase the count of another integer value within your item's `use` method -- say, to increase the number of times it performs some repeated effect.
+
+First, invoke one of the overloads of `EnchantmentHelper#runIterationOnItem`. This function accepts an `EnchantmentHelper.EnchantmentVisitor`, which is a functional interface that accepts an enchantment and its level, and is invoked on all of the enchantments that the given itemstack has (essentially a `BiConsumer<Enchantment, Integer>`). While any consumer of those two values will work, the `Enchantment` class provides a handy function that fits this interface (and is used often by vanilla) -- `Enchantment#applyEffects`. This method is used as above to test the conditions of the `ConditionalEffect`s.
+
+To actually perform the adjustment, use the provided `Increment#add` method.
+
+Note that in this example, the level of the enchantment does not affect the outcome. This can be changed by using `enchantLevel` somewhere in the `Consumer<T>` lambda expression (the last line with code in the example). Any other information stored in the `ItemStack` can also be accessed from here, so other Data Components could be used to inform how the adjustment goes.
```

Comment:
This is not true, as the condition that would be applied could use the enchantment level. You are simply not using it within the actual enchantment effect object.